gen-ai/orchestrator-server/src/main/python/server/src/gen_ai_orchestrator/services/langchain/rag_chain.py

```diff
@@ -99,10 +100,22 @@ async def execute_qa_chain(query: RagQuery, debug: bool) -> RagResponse:
         'RAG chain - Use RetrieverJsonCallbackHandler for debugging : %s',
         debug,
     )
+
+    callback_handlers = []
     records_callback_handler = RetrieverJsonCallbackHandler()
+    if debug:
+        # Debug callback handler
+        callback_handlers.append(records_callback_handler)
+    if query.observability_setting is not None:
+        # Langfuse callback handler
+        callback_handlers.append(
+            create_langfuse_callback_handler(
```

Comment:
This function should clearly be renamed, langfuse is a specific observability provider name : 
```suggestion
            create_observability_callback_handler(
```